### PR TITLE
feat: enabled support for background image and color

### PR DESCRIPTION
**Pull request by chinmay-sg**
## Motivation

- Added support for background color in nucleus/src/components/Cell.jsx
- Added support for background image in nucleus/src/components/Supernova.jsx

### CODE REVIEW

File: apis/nucleus/src/components/Cell.jsx [hunk 1]
```diff
@@ -297,6 +297,19 @@ const Cell = forwardRef(
...
 
+    let bgColor = 'transparent';
+
+    if (
+      layout &&
+      layout.components &&
+      layout.components.find((comp) => comp.key === 'general') &&
+      layout.components.find((comp) => comp.key === 'general').bgColor &&
+      layout.components.find((comp) => comp.key === 'general').bgColor.color &&
+      layout.components.find((comp) => comp.key === 'general').bgColor.color.color
```
Comment (Caele) [inline apis/nucleus/src/components/Cell.jsx:308]: simplify this:
const component = layout?.components?.find((comp) => comp.key === 'general');
if(component?.bgColor?.color) {
...
}
I think you can use the ? notation here, but if not you can at least stop it from running .find() 4 times!

File: apis/nucleus/src/components/Supernova.jsx [hunk 2]
```diff
@@ -119,11 +119,55 @@ function Supernova({ sn, snOptions: options, snPlugins: plugins, layout, appLayo
...
-      style={{ position: 'relative', height: '100%' }}
+      style={{
+        position: 'relative',
+        height: '100%',
+        backgroundImage: `url(${
+          layout &&
+          layout.components &&
+          layout.components.find((comp) => comp.key === 'general') &&
+          layout.components.find((comp) => comp.key === 'general').bgImage &&
+          layout.components.find((comp) => comp.key === 'general').bgImage.mediaUrl
```
Comment (Caele) [inline apis/nucleus/src/components/Supernova.jsx:165]: Same here, way too verbose. Save the varaible above it instead.

File: apis/nucleus/src/components/Cell.jsx (hunk 1 above)
Comment (chinmay-sg) [reply apis/nucleus/src/components/Cell.jsx:308]: The thing is that for some reason, lint is not allowing to commit with the ? notation, it's throwing the following error.
 
Screenshot 2022-09-06 at 3 12 05 PM

I had done the same thing in a single line earlier, but it didn't allow me to commit

File: apis/nucleus/src/components/Supernova.jsx [hunk 3]
```diff
@@ -119,11 +119,55 @@ function Supernova({ sn, snOptions: options, snPlugins: plugins, layout, appLayo
...
+    fitWidth: '100% auto',
+    fitHeight: 'auto 100%',
+    stretchFit: '100% 100%',
+    alwaysFill: 'cover',
+  };
+
+  function getBackgroundPosition(data) {
+    let bkgImagePosition = 'center center';
+    if (data && data.components && data.components.length > 0 && data.components !== 'none') {
+      const styling = data.components.find((comp) => comp.key === 'general');
```
Comment (Caele) [inline apis/nucleus/src/components/Supernova.jsx:134]: again, this "finding" of the component should be done once per layout change.

Comment (chinmay-sg) [reply apis/nucleus/src/components/Supernova.jsx:134]: ok, will change this

File: apis/nucleus/src/components/Cell.jsx (hunk 1 above)
Comment (Caele) [reply apis/nucleus/src/components/Cell.jsx:308]: ok, but you can atlest simplify it to:
const component = layout &&  layout.components ? layout.components.find((comp) => comp.key === 'general') : null;

File: apis/nucleus/src/components/Cell.jsx [hunk 4]
```diff
@@ -297,6 +297,19 @@ const Cell = forwardRef(
...
+    ) {
+      bgColor = layout.components.find((comp) => comp.key === 'general').bgColor.color.color;
+    }
```
Comment (Caele) [inline apis/nucleus/src/components/Cell.jsx:311]: Also, don't you need to go through the theme to resolve the color?
Some thing like:
halo.public.theme.getColorPickerColor(bgColor.color)
I assume this is done in the client, but in a slightly different way?

File: apis/nucleus/src/components/Cell.jsx (hunk 1 above)
Comment (chinmay-sg) [reply apis/nucleus/src/components/Cell.jsx:308]: ok will do that

**Pull request closed without merging**